### ISSUE TITLE
Feature Request: evcc support

### ISSUE THREAD

**Issue by saimonsez** [yoziru/esphome-tesla-ble#99]
Hi there

It would be awesome if you could provide support for evcc: https://github.com/evcc-io/evcc. There is another project with this feature set: https://github.com/wimaha/TeslaBleHttpProxy I‘m experiencing reliability issues on this which I believe to be caused by go-ble.

**Comment by rogervdh** [comment]
Would also love EVCC support. Thanks for considering.

**Comment by wijnands66** [comment]
Would be great! Especially for Tesla Wallbox users. Thanks for considering.

**Comment by danielraffel** [comment]
If useful, I switched to using TeslaBLEHttpProxy with EVCC (and my UMC) and shared how. https://danielraffel.me/2025/01/03/how-i-automated-ev-charging-using-evcc-an-eg4-solar-inverter-and-a-tesla-mobile-connector/

**Comment by Trunks1982** [comment]
Hi wäre cool aber da fehlt noch der SOC und Ober Ladestecker eingesteckt ist.

**Comment by top-gun** [comment]
> MQTT works perfectly with ESPHome. My complain is, that you need MQTT broker to work with it, it's easier than full blown HomeAssistant, but still additional point of failure
> 
> I would prefer a HTTP API on ESPHome device side, so EVCC can talk to ESPHome device directly

When you activate the web server on the device, you can access every entity via http. 
https://esphome.io/web-api/

An example for the energy counter from my wall charger: 
<img width="1804" height="324" alt="Image" src="https://github.com/user-attachments/assets/52f2a392-bd5b-433f-a5b4-4b6f8f94bf6d" />
The API allows access of any entity, you can switch switches, open doors etc.

Fwiw, having an MQTT broker in the network is useful for a lot of cases. EVCC comes with Mosquitto out of the box. I prefer to use the one on my Home Assistant server, though, as it's generally advisable to have only one MQTT broker that centralises all topics.

**Comment by top-gun** [comment]
> can you also control via this web server? (I'know about it, but didn't tested it outside of "Web UI")

Yes. You can query and control all entities. Please have a look at the page I linked, it‘s a very powerful api, but it needs some reading. Data is generally returned in Json format, and switches etc are controlled with keywords that get added to the url.

**Comment by yoziru** [comment]
**Evidence**: evcc support would require exposing charge state data via MQTT or HTTP endpoints that evcc can consume. This is an integration concern that belongs at the Home Assistant layer (where the data is already available via sensors), not in the ESP32 firmware itself.

**Analysis**: evcc already integrates with Home Assistant via the EVCC addon. All charging sensors (battery level, charger power, charging amps, etc.) are exposed through the native ESPHome API to HA. evcc can be configured to read from HA directly.

**Resolution**: Out of scope for this firmware. evcc integration should be done via Home Assistant. Closing.
